classes/invitation/core/Invitation.php

```diff
@@ -284,4 +468,30 @@ protected function getExpiryDays(): int
     {
         return (int) Config::getVar('invitations', 'expiration_days', self::DEFAULT_EXPIRY_DAYS);
     }
+
+    /**
+     * this function is overriden by custom invitations if necessary
+     * so that properties of the invitation are filled before returning 
+     * the invitation object to the code.
+     * Used in InvitationFactory::getExisting.
+     */
+    public function fillCustomProperties(): void
+    {
+        return;
+    }
+
+    public function getUserId(): ?int
```

Comment:
Since the `invitationModel` is a public attribute anyway, I'd rather we reduced the number of getters in favour of just accessing that directly.